master/docs/developer/utils.rst

```diff
@@ -886,6 +886,44 @@ For example, a particular daily scheduler could be configured on multiple master
         Therefore, in this method it is safe to reassign the "active" status to another instance.
         This method may return a Deferred.
 
+.. py:class:: SingletonService
+
+    This class implements a generic Service that needs to be instantiated as a singleton according to its parameters.
```

Comment:
I think you are right, we need to change the name
http://stackoverflow.com/questions/1050991/singleton-with-arguments-in-java